### DISCUSSION TITLE
"... Only TXT files are allowed."

### DISCUSSION THREAD

**Discussion by lorenzo4519** [Quansight/ragna#206] [Q&A] [answered]
Hello,

I'm using ragna ui with openai on windows 10 and it works great.
The only issue is that I'm not able to pass PDF files to the UI, but only TXT.
In the GUI in the area for drag and drop the following text is displayed:
"click to upload or drag and drop. Only TXT files are allowed."
I tried to pass PDF files anyway but it doesn't work and produces the following errors in CLI:
INFO:     127.0.0.1:57022 - "POST /document HTTP/1.1" 500 Internal Server Error
INFO:     127.0.0.1:57024 - "POST /chats HTTP/1.1" 500 Internal Server Error

I started from a python 3.11 installation and on top I installed with pip just openai, "ragna[all]", jupiterlab.
I confirmed pymupdf version 1.23.6 is installed. further details below

It would be great if you could help me

Thank you in advance
Lorenzo

**pip freeze**
aiofiles==23.2.1
aiohttp==3.8.6
aiosignal==1.3.1
annotated-types==0.6.0
anyio==3.7.1
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asttokens==2.4.1
async-lru==2.0.4
async-timeout==4.0.3
attrs==23.1.0
Babel==2.13.1
backoff==2.2.1
bcrypt==4.0.1
beautifulsoup4==4.12.2
bleach==6.1.0
bokeh==3.3.1
cachetools==5.3.2
certifi==2023.7.22
cffi==1.16.0
charset-normalizer==3.3.2
chroma-hnswlib==0.7.3
chromadb==0.4.17
click==8.1.7
colorama==0.4.6
coloredlogs==15.0.1
comm==0.2.0
contourpy==1.2.0
debugpy==1.8.0
decorator==5.1.1
defusedxml==0.7.1
Deprecated==1.2.14
deprecation==2.1.0
distro==1.8.0
emoji==2.8.0
executing==2.0.1
fastapi==0.104.1
fastjsonschema==2.19.0
filelock==3.13.1
flatbuffers==23.5.26
fqdn==1.5.1
frozenlist==1.4.0
fsspec==2023.10.0
google-auth==2.23.4
googleapis-common-protos==1.61.0
greenlet==3.0.1
grpcio==1.59.2
h11==0.14.0
httpcore==1.0.2
httptools==0.6.1
httpx==0.25.1
huey==2.5.0
huggingface-hub==0.19.3
humanfriendly==10.0
idna==3.4
importlib-metadata==6.8.0
importlib-resources==6.1.1
ipykernel==6.26.0
ipython==8.17.2
isoduration==20.11.0
jedi==0.19.1
Jinja2==3.1.2
json5==0.9.14
jsonpointer==2.4
jsonschema==4.19.2
jsonschema-specifications==2023.11.1
jupyter-events==0.9.0
jupyter-lsp==2.2.0
jupyter_client==8.6.0
jupyter_core==5.5.0
jupyter_server==2.10.1
jupyter_server_terminals==0.4.4
jupyterlab==4.0.8
jupyterlab-pygments==0.2.2
jupyterlab_server==2.25.1
kubernetes==28.1.0
lancedb==0.3.3
linkify-it-py==2.0.2
Markdown==3.5.1
markdown-it-py==3.0.0
MarkupSafe==2.1.3
matplotlib-inline==0.1.6
mdit-py-plugins==0.4.0
mdurl==0.1.2
mistune==3.0.2
monotonic==1.6
mpmath==1.3.0
multidict==6.0.4
nbclient==0.9.0
nbconvert==7.11.0
nbformat==5.9.2
nest-asyncio==1.5.8
notebook_shim==0.2.3
numpy==1.26.2
oauthlib==3.2.2
onnxruntime==1.16.2
openai==1.3.0
opentelemetry-api==1.21.0
opentelemetry-exporter-otlp-proto-common==1.21.0
opentelemetry-exporter-otlp-proto-grpc==1.21.0
opentelemetry-proto==1.21.0
opentelemetry-sdk==1.21.0
opentelemetry-semantic-conventions==0.42b0
overrides==7.4.0
packaging==23.2
pandas==2.1.3
pandocfilters==1.5.0
panel==1.3.1
param==2.0.1
parso==0.8.3
Pillow==10.1.0
platformdirs==4.0.0
posthog==3.0.2
prometheus-client==0.18.0
prompt-toolkit==3.0.36
protobuf==4.25.1
psutil==5.9.6
pulsar-client==3.3.0
pure-eval==0.2.2
py==1.11.0
pyarrow==14.0.1
pyasn1==0.5.0
pyasn1-modules==0.3.0
pycparser==2.21
pydantic==2.5.1
pydantic-settings==2.1.0
pydantic_core==2.14.3
Pygments==2.16.1
PyJWT==2.8.0
pylance==0.8.10
PyMuPDF==1.23.6
PyMuPDFb==1.23.6
PyPika==0.48.9
pyreadline3==3.4.1
python-dateutil==2.8.2
python-dotenv==1.0.0
python-json-logger==2.0.7
python-multipart==0.0.6
pytz==2023.3.post1
pyviz_comms==3.0.0
pywin32==306
pywinpty==2.0.12
PyYAML==6.0.1
pyzmq==25.1.1
questionary==2.0.1
Ragna==0.1.1
ratelimiter==1.2.0.post0
redis==5.0.1
referencing==0.31.0
regex==2023.10.3
requests==2.31.0
requests-oauthlib==1.3.1
retry==0.9.2
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==13.7.0
rpds-py==0.13.0
rsa==4.9
semver==3.0.2
Send2Trash==1.8.2
six==1.16.0
sniffio==1.3.0
soupsieve==2.5
SQLAlchemy==2.0.23
stack-data==0.6.3
starlette==0.27.0
sympy==1.12
tenacity==8.2.3
terminado==0.18.0
tiktoken==0.5.1
tinycss2==1.2.1
tokenizers==0.15.0
tomlkit==0.12.3
tornado==6.3.3
tqdm==4.66.1
traitlets==5.13.0
typer==0.9.0
types-python-dateutil==2.8.19.14
typing_extensions==4.8.0
tzdata==2023.3
uc-micro-py==1.0.2
uri-template==1.3.0
urllib3==1.26.18
uvicorn==0.24.0.post1
watchfiles==0.21.0
wcwidth==0.2.10
webcolors==1.13
webencodings==0.5.1
websocket-client==1.6.4
websockets==12.0
wrapt==1.16.0
xyzservices==2023.10.1
yarl==1.9.2
zipp==3.17.0

**Comment by pmeier** [comment, marked as answer]
Hey @lorenzo4519 :wave:

Could you run the following for me

```bash
$ python -c 'from ragna.core import PdfDocumentHandler; print(PdfDocumentHandler.is_available())'
```

If that is `False` something is wrong with your environment. If it is `True` however, could you run the following script when the UI (or just the API) is running and post the output

```python
import httpx

client = httpx.Client(base_url="http://127.0.0.1:31476")
client.get("/").raise_for_status()

token = (
    client.post(
        "/token",
        data={
            "username": "ragna-206",
            "password": "ragna-206",
        },
    )
    .raise_for_status()
    .json()
)
client.headers["Authorization"] = f"Bearer {token}"

components = client.get("/components").raise_for_status().json()
print(components["documents"])
```

**Comment by lorenzo4519** [comment]
Thank you @pmeier for your quick feedback!

the first command took a few seconds and returned false.
I reset the environment and now it works.

**Comment by AnselmJeong** [comment]
How did you **reset** your environment?
I've met the very same phenomenon, and `PdfDocumentHandler.is_available()` said `False`

**Reply by pmeier** [reply]
> I've met the very same phenomenon

How did you install Ragna? `pip install ragna` or `pip install 'ragna[all]'`? Did you check if you have PyMuPDF correctly installed? You can check by running 

```shell
$ python -c 'import fitz; print(fitz.__version__)'
```

If that fails, either install it manually with `pip install pymupdf` or install Ragna with all optional packages, i.e. `pip install 'ragna[all]'`. Afterwards, the command above should show you the version number of the installed package. Now also rerun

```shell
$ python -c 'from ragna.core import PdfDocumentHandler; print(PdfDocumentHandler.is_available())'
```

and that should come out as "True". Only if that is not the case, proceed with the steps below.

> How did you **reset** your environment?

Depends. If you are running in a virtual environment, just delete it and create a new one. If you are using your system Python and thus have installed Ragna as a system package, resetting the environment is basically the same as resetting the system. In that case it is probably a lot better to find the root cause for the error and fix it rather than start from scratch.
